### PR TITLE
Issue #382: Add permissions for taxonomy on the vocabulary form.

### PR DESCRIPTION
**Pull request by jenlampton**
PR for https://github.com/backdrop/backdrop-issues/issues/38

### CODE REVIEW

File: core/modules/taxonomy/taxonomy.module [hunk 1]
```diff
@@ -37,17 +37,31 @@ function taxonomy_permission() {
...
+ *
+ * @return array
+ *   An array of permission names and descriptions.
+ */
+function taxonomy_vocabulary_list_permissions($vocabulary) {
+  $permissions = array(
+    'edit terms in ' . $vocabulary->machine_name => array(
+      'title' => t('%vocabulary: Edit terms', array('%vocabulary' => $vocabulary->name)),
+    ),
+     'delete terms in ' . $vocabulary->machine_name => array(
```
Comment (quicksketch) [inline core/modules/taxonomy/taxonomy.module]: Extra space here before `'delete`

**Comment by docwilmot** [comment]
Permissions are not actually saving:
- go to 'tag' config at `admin/structure/taxonomy/tags/configure`, note current permissions set A
- click checkboxes and save
- go back to `admin/structure/taxonomy/tags/configure`, permissions still at A
- go to `/permissions` change tags vocab to B, it works
- go to `admin/structure/taxonomy/tags/configure`, perms now reflect B.

**Comment by docwilmot** [comment]
Doesn't seem to be any code in `taxonomy_form_vocabulary_submit()` to save perms?

**Comment by jenlampton** [comment]
Fixed the extra whitespace, and the 'No permissions set' text in the vertical tab.

**Comment by quicksketch** [comment]
Last merge before 1.3.0. Thanks!!

**Pull request merged**